development/models/randomForest.py

```diff
@@ -0,0 +1,177 @@
+# Various imports used for notebook
+import pandas as pd
+import seaborn as sns
+from sklearn.metrics import confusion_matrix
+from sklearn.metrics import classification_report
+from sklearn.ensemble import RandomForestClassifier
+from sklearn.model_selection import train_test_split
+from sklearn.metrics import accuracy_score
+from sklearn.tree import plot_tree
+import matplotlib.pyplot as plt
+import joblib
+from sklearn.metrics import f1_score
+from sklearn.model_selection import GridSearchCV
+from sklearn.feature_extraction.text import TfidfVectorizer
+
+# Read cleaned dataset
+data = pd.read_sql_query("SELECT * FROM 'mortgage complaints'", "sqlite:///StaterData.db")
+
+# Limit test size due to performance issues
+data = data.loc[:1000]
+
+# Define your features and target
+X = data['Consumer complaint narrative']  # text_column is the name of the column in your dataset that contains the text data
+y = data['Issue']  # target_column is the name of the column in your dataset that contains the target variable
+
+# Split your data into training and testing sets
+X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.3, random_state=2)
+
+# Define params from GridSearch which performed the best
+param_grid = {
+    'n_estimators': 200,
+    'max_depth': None,
+    'min_samples_split': 2,
+    'min_samples_leaf': 1,
+}
+
+# Vectorize your text data using TF-IDF vectorizer
+vectorizer = TfidfVectorizer(stop_words='english')
+X_train = vectorizer.fit_transform(X_train)
+X_test = vectorizer.transform(X_test)
+
+# Create a Random Forest model
+rf_model = RandomForestClassifier(random_state=2)
+
+# Set the params for the model
+rf_model.set_params(**param_grid)
+
+# Train the model on the training data
+rf_model.fit(X_train, y_train)
+
+# Predict the target variable for the testing data
+y_pred = rf_model.predict(X_test)
+
+# Evaluate the accuracy of the model
+accuracy = accuracy_score(y_test, y_pred)
+print('Accuracy:', accuracy)
+f1score = f1_score(y_test, y_pred, average='macro')
+print('F1 score:', f1score)
+
+custom_text = "My mortgage was sold to Roundpoint company in XX/XX/2023. I got a letter in XXXX from my prior lender, XXXX XXXX stating this was to occur and the next payment to be made would go to Roundpoint. X..."
+custom_text_bow = vectorizer.transform([custom_text])
+
```

Comment:
Should be in a function wich can be callable and has an parameter called custom_text so the chatbot can call it. Function should return the category that is predicted

Reply:
committed an callable function please check if it's okay